src/config.ml

```diff
@@ -146,10 +146,19 @@ module Conf_map = struct
          | None | Some `Server -> Error "is not a TLS client"
          | Some `Client -> Ok ()) >>= fun () ->
         let _todo = ensure_not in
-        ensure_mem Auth_user_pass "does not have user/password"
-        (* ^-- TODO or has client certificate ? see options.c:--cert/--key,--pkcs12,or--auth-user-pass *)
+        begin match find Auth_user_pass t, find Tls_cert t, find Tls_key t with
+          | Some _, None, None | None, Some _, Some _ -> Ok ()
+          | None, None, None ->
+            Error "does not have user/password, neither TLS certificate"
+          | Some _, Some _, Some _  ->
+            Error "both user/pass and TLS certificate provided"
+          | _ ->
+            Error "invalid combination of user/password and TLS key and certificate"
+          (* ^-- TODO or has -pkcs12 *)
+        end >>= fun () ->
+        ensure_mem Cipher "client must specify 'cipher AES-256-CBC'"
         >>= fun () ->
-        (if mem Cipher t && get Cipher t <> "AES-256-CBC" then
+        (if get Cipher t <> "AES-256-CBC" then
```

Comment:
This (`if get Cipher t <> "AES-256-CBC" then`) now raises an exception if there is no `Cipher`.
With this patch that is currently guaranteed by the `ensure_mem` above, but openvpn files do not require `cipher` (they will pick a default somehow, for clients I think they use the cipher picked by the server in this case), so I think we should keep the two separate.
The reason for having two checks is that `current only supported Cipher is ...` is a bit confusing when the user has no `cipher` in their original config file from OpenVPN.

Reply:
ok, reverted this change in d129dfc